webviz_subsurface/plugins/_group_tree/group_tree.py

```diff
@@ -0,0 +1,201 @@
+        smry["DATE"] = pd.to_datetime(smry["DATE"])  # .dt.date
+        gruptree["DATE"] = pd.to_datetime(gruptree["DATE"]).dt.date
+
+        if time_index == "yearly":
+            smry = smry[smry["DATE"].dt.is_year_start]
+        smry["DATE"] = smry["DATE"].dt.date
+
+        self.grouptreedata = GroupTreeData(smry, gruptree)
+
+        self.set_callbacks(app)
+
+    def add_webvizstore(self) -> List[Tuple[Callable, List[Dict]]]:
+        functions: List[Tuple[Callable, List[Dict]]] = self.emodel.webvizstore
+        functions.append(
+            (
+                read_gruptree_files,
+                [{"ens_paths": self.ens_paths, "gruptree_file": self.gruptree_file}],
+            )
+        )
+        return functions
+
+    @property
+    def tour_steps(self) -> List[dict]:
+        return [
+            {
+                "id": self.uuid("layout"),
+                "content": "Dashboard vizualizing Eclipse network tree.",
+            },
+            {
+                "id": self.uuid("selections_layout"),
+                "content": "Menu for selecting ensemble and other options.",
+            },
+            {
+                "id": self.uuid("filters_layout"),
+                "content": "Menu for filtering options.",
+            },
+            {
+                "id": self.uuid("grouptree_wrapper"),
+                "content": "Vizualisation of network tree.",
+            },
+        ]
+
+    @property
+    def layout(self) -> html.Div:
+        return html.Div(
+            children=[
+                # clientside_stores(get_uuid=self.uuid),
+                main_view(get_uuid=self.uuid, ensembles=self.ensembles),
+            ],
+        )
+
+    def set_callbacks(self, app: dash.Dash) -> None:
+        controllers(app=app, get_uuid=self.uuid, grouptreedata=self.grouptreedata)
+
+
+@webvizstore
+def read_gruptree_files(ens_paths: Dict[str, str], gruptree_file: str) -> pd.DataFrame:
+    """Searches for gruptree files on the scratch disk. These
+    files can be exported in the FMU workflow using the ECL2CSV
+    forward job with subcommand gruptree.
+    """
+    df = pd.DataFrame()
+    for ens_name, ens_path in ens_paths.items():
+        df_ens = read_ensemble_gruptree(ens_name, ens_path, gruptree_file)
+        df_ens["ENSEMBLE"] = ens_name
+        df = pd.concat([df, df_ens])
+    return df.where(pd.notnull(df), None)
+
+
+def read_ensemble_gruptree(
+    ens_name: str, ens_path: str, gruptree_file: str
+) -> pd.DataFrame:
+    """Reads the gruptree file for and ensemble.
+
+    If BRANPROP is found in the KEYWORD column, then GRUPTREE rows
+    are filtered out.
+
+    If the trees are equal in every realization, only one realization is kept.
+    """
+    ens = ScratchEnsemble("ens", ens_path)
+    df_files = ens.find_files(gruptree_file)
+
+    if df_files.empty:
+        raise ValueError(f"No gruptree file available for ensemble: {ens_name}")
+
+    # Load all gruptree dataframes and check if they are equal
```

Comment:
Should this dataframe comparison be refactored out into its own testable function?

Reply:
Yes, good idea. I don't have time to do it now so I will make an issue.

Reply:
https://github.com/equinor/webviz-subsurface/issues/790